src/reporting.cc

```diff
@@ -0,0 +1,164 @@
+#include "reporting.h"
+
+#include <algorithm>
+#include <iostream>
```

Comment:
hmm.. why are these `includes` before the comment header? Other files have them after the comment header.